### ISSUE TITLE
Form model addRule() data property undefined

### ISSUE THREAD

**Issue by lmovse** [rsuite/rsuite#288]
### Bug, feature request, or proposal:
Bug

### What is the expected behavior?
当我为某一 form 字段的校验添加自定义规则时，应该可以读取 data 属性，然后根据 data 里面的其他字段的值动态判断当前字段的值是否合法，下面的代码摘自官方文档 Form 里的一个 CheckForm 的例子
```js
const model = Schema.Model({
  name: StringType().isRequired('This field is required.'),
  email: StringType()
    .isEmail('Please enter a valid email address.')
    .isRequired('This field is required.'),
  age: NumberType('Please enter a valid number.').range(
    18,
    30,
    'Please enter a number from 18 to 30'
  ),
  password: StringType().isRequired('This field is required.'),
  verifyPassword: StringType()
    .addRule((value, data) => {
      console.log(data);

      if (value !== data.password) {
        return false;
      }

      return true;
    }, 'The two passwords do not match')
    .isRequired('This field is required.')
});
```

### What is the current behavior?
获取不到 `addRule((value, data) => ..., '')` 里的 data 的值，提示该值为 undefined

### What are the steps to reproduce?
将官方文档里的 Form 一节的表单数据校验小节里面的 CheckForm  组件 copy 到项目里，然后渲染该组件，在 `verifyPassword` 一栏输入时会直接报错，报错如下

```js
CheckForm.js:30 Uncaught TypeError: Cannot read property 'password' of undefined
    at CheckForm.js:30
    at StringType.check (Type.js:54)
    at Schema.checkForField (Schema.js:32)
    at DefaultPropsComponent.handleFieldCheck (FormControl.js:183)
    at handleFieldChange (FormControl.js:161)
```

**Comment by simonguo** [comment]
@lmovse  你的版本是？

**Comment by lmovse** [comment]
> @lmovse 你的版本是？

3.2.1

**Comment by simonguo** [comment]
这个功能是在 [3.2.8](https://github.com/rsuite/rsuite/releases/tag/3.2.8) 加入的。
建议你升级 rsuite 到最新版本

**Comment by lmovse** [comment]
OK，可以了.....抱歉，是我的疏忽，麻烦你了

**Comment by matepaiva** [comment]
Hello, sorry. I am having this exact same problem, but I am with `"rsuite": "^4.8.4"`. Do you have any idea about why I am getting this error?

**Comment by matepaiva** [comment]
Already found my solution. I was missing to pass the `formValue` to `<Form />`. Thank you all.

**Comment by mrskhris** [comment]
Rsuite version 5.70.3 (but checked 5.70.2 too). I am passing form value, but having same issue: no data shows up in addRule.

**Comment by myNameIsDu** [comment]
> Rsuite version 5.70.3 (but checked 5.70.2 too). I am passing form value, but having same issue: no data shows up in addRule.

<img width="1278" alt="image" src="https://github.com/user-attachments/assets/bff98845-20c8-4c88-b50d-dbddebe28817">


I tried it on the latest version(5.70.3) and it works. Could you please provide steps to reproduce or an online minimal reproduction?

**Comment by mrskhris** [comment]
Thank you for taking a look, @myNameIsDu. It seems I did not describe the problem in full. Yes, full form data is logged if we start changing contents of e-mail field. But what happens when we try to submit the form, during final check? For me no data is shown, only the value of e-mail field.
Here is the example:
https://codesandbox.io/p/sandbox/zealous-ives-46ghtc?workspaceId=b01acaa5-8239-4931-9e84-dd360a812b7c

**Comment by simonguo** [comment]
Hi @mrskhris 

Fixed in rsuite@5.71.0

**Comment by mrskhris** [comment]
Thank you so much!
